src/Sql/dbo/Stored Procedures/Organization_EnableCollectionEnhancements.sql

```diff
@@ -0,0 +1,160 @@
+                    LEFT JOIN [dbo].[CollectionGroup] CG ON CG.[CollectionId] = C.[Id] AND CG.[GroupId] = TG.[GroupId]
+                    WHERE CG.[CollectionId] IS NULL;
+
+                    -- Update Group to clear AccessAll flag
+                    UPDATE G
+                    SET [AccessAll] = 0
+                    FROM [dbo].[Group] G
+                    INNER JOIN #TempStep1 TG ON G.[Id] = TG.[GroupId];
+
+                -- Step 2
+                    -- Update existing rows in [dbo].[CollectionUser]
+                    UPDATE target
+                    SET
+                        target.[ReadOnly] = 0,
+                        target.[HidePasswords] = 0,
+                        target.[Manage] = 0
+                        FROM [dbo].[CollectionUser] AS target
+                        INNER JOIN (
+                            SELECT C.[Id] AS [CollectionId], T.[OrganizationUserId]
+                            FROM [dbo].[Collection] C
+                            INNER JOIN #TempStep2 T ON C.[OrganizationId] = T.[OrganizationId]
+                        ) AS source
+                    ON target.[CollectionId] = source.[CollectionId] AND target.[OrganizationUserId] = source.[OrganizationUserId];
+
+                    -- Insert new rows into [dbo].[CollectionUser]
+                    INSERT INTO [dbo].[CollectionUser] ([CollectionId], [OrganizationUserId], [ReadOnly], [HidePasswords], [Manage])
+                    SELECT source.[CollectionId], source.[OrganizationUserId], 0, 0, 0
+                    FROM (
+                             SELECT C.[Id] AS [CollectionId], T.[OrganizationUserId]
+                             FROM [dbo].[Collection] C
+                                 INNER JOIN #TempStep2 T ON C.[OrganizationId] = T.[OrganizationId]
+                         ) AS source
+                    LEFT JOIN [dbo].[CollectionUser] AS target
+                        ON target.[CollectionId] = source.[CollectionId] AND target.[OrganizationUserId] = source.[OrganizationUserId]
+                    WHERE target.[CollectionId] IS NULL;
+
+                    -- Update OrganizationUser to clear AccessAll flag
+                    UPDATE OU
+                    SET [AccessAll] = 0
+                    FROM [dbo].[OrganizationUser] OU
+                    INNER JOIN #TempStep2 T ON OU.[Id] = T.[OrganizationUserId];
+
+                -- Step 3
+                    -- Update [dbo].[CollectionUser] with [Manage] = 1 using the temporary table
+                    UPDATE cu
+                    SET cu.[ReadOnly] = 0,
+                        cu.[HidePasswords] = 0,
+                        cu.[Manage] = 1
+                    FROM [dbo].[CollectionUser] cu
+                    INNER JOIN #TempStep3 temp ON cu.[OrganizationUserId] = temp.[OrganizationUserId];
+
+                    -- Insert rows to [dbo].[CollectionUser] with [Manage] = 1 using the temporary table
```

Comment:
I'd like a bit more explanation here:

```suggestion
                    -- Insert rows to [dbo].[CollectionUser] with [Manage] = 1 using the temporary table
                    -- This is for orgUsers who are Managers / EditAssignedCollections but have access via a group
                    -- We cannot give the whole group Manage permissions so we have to give them a direct assignment
```